File: app/src/main/java/no/nav/sosialhjelp/soknad/consumer/virusscan/ScanResult.java
package no.nav.sosialhjelp.soknad.consumer.virusscan;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

class ScanResult {
    @JsonAlias("Filename")
    private final String filename;
    @JsonAlias("Result")
    private final Result result;

    @JsonCreator
    ScanResult(@JsonProperty("filename") String filename, @JsonProperty("result") Result result) {
        this.filename = filename;
        this.result = result;
    }

    public String getFilename() {
        return filename;
    }

    public Result getResult() {
        return result;
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/vedlegg/virusscan/dto/ScanResult.kt
package no.nav.sosialhjelp.soknad.vedlegg.virusscan.dto

import com.fasterxml.jackson.annotation.JsonAlias

data class ScanResult(
    @param:JsonAlias("Filename")
    val filename: String?,
    @param:JsonAlias("Result")
    val result: Result,
)

enum class Result {
    FOUND,
    OK,
    ERROR,
}
